mwdb/web/src/components/ShowObject/Views/RelatedFilesTab.js

```diff
@@ -0,0 +1,204 @@
+    useEffect(() => {
+        getRelatedFiles(api, context);
+        // eslint-disable-next-line
+    }, [getRelatedFiles]);
+
+    if (!context.object.related_files) {
+        return <div class="card-body text-muted">Loading...</div>;
+    }
+    if (context.object.related_files.length === 0) {
+        return <div class="card-body text-muted">Nothing to show here</div>;
+    }
+
+    return (
+        <table className="table table-striped table-bordered table-hover data-table">
+            <thead>
+                <th>File name</th>
+                <th>File size</th>
+                <th>Download</th>
+                <th>Remove</th>
+            </thead>
+            {context.object.related_files.map((related_file) => (
+                <RelatedFileItem {...related_file} />
+            ))}
+        </table>
+    );
+}
+
+export default function RelatedFilesTab() {
+    const [showModal, setShowModal] = useState();
+    const [file, setFile] = useState(null);
+    const context = useContext(ObjectContext);
+    const api = useContext(APIContext);
+    const { setObjectError } = context;
+
+    const modalStyle = {
+        content: {
+            top: "50%",
+            left: "50%",
+            right: "auto",
+            bottom: "auto",
+            marginRight: "-50%",
+            transform: "translate(-50%, -50%)",
+        },
+    };
+
+    async function handleSubmit() {
+        try {
+            await api.uploadRelatedFile(file, context.object.sha256);
+            updateRelatedFiles(api, context);
+        } catch (error) {
+            setObjectError(error);
+        }
+    }
+
+    return (
+        <div>
+            <ObjectTab
+                tab="related_files"
+                label="Related files"
+                icon={faExternalLinkSquare}
+                component={ShowRelatedFiles}
+                actions={[
+                    <ObjectAction
+                        label="Upload new"
+                        icon={faPlus}
+                        action={() => {
+                            setShowModal(true);
+                        }}
+                    />,
+                ]}
+            />
+            <ReactModal
+                isOpen={showModal}
+                onRequestClose={() => {
+                    setShowModal(false);
+                }}
+                style={modalStyle}
+            >
+                <form
+                    name="RelatedFileUploadForm"
+                    onSubmit={() => {
+                        handleSubmit();
+                        setShowModal(false);
+                    }}
+                >
+                    <input
+                        class="modal-header"
+                        name="RelatedFileUploadField"
+                        type="file"
+                        required="required"
+                        onChange={() =>
+                            setFile(
+                                document.forms["RelatedFileUploadForm"][
```

Comment:
I would use React refs here instead of direct DOM access
- https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element
- https://reactjs.org/docs/refs-and-the-dom.html#refs-and-function-components

Reply:
I changed my code to use event